docs/specification.md

```diff
@@ -1,150 +1,371 @@
 
+### F5. The GA4GH digest algorithm
 
-### Sequence collection as input
+The GA4GH digest algorithm, `sha512t24u`, was created as part of the [Variation Representation Specification standard](https://vrs.ga4gh.org/en/stable/impl-guide/computed_identifiers.html).  This procedure is described as ([Hart _et al_. 2020](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0239883)):
 
-* **sequences -> refget digests**: Individual sequences can be converted into refget digests using the refget algorithm
-* **sequences -> seqcol digest**: Sequence collections can be converted into seqcol digests using the seqcol algorithm. To obtain a sequence digest, follow the algorithm outlined above, sequences -> refget digests -> seqcol digest.
+- performing a SHA-512 digest on a binary blob of data
+- truncate the resulting digest to 24 bytes
+- encodes the 24 bytes using `base64url` ([RFC 4648](https://datatracker.ietf.org/doc/html/rfc4648#section-5)) resulting in a 32 character string
 
-### Refget digest as input
+In Python, the digest can be computed with this function:
 
-* **refget digest -> seqcol digests containing this sequence**: Use the `containing_collections` secondary endpoint. 
-* **many refget digests -> seqcol digests containing all of these sequences**: Repeated application of `containing_collections` endpoint?
+```python
+def trunc512_digest(seq, offset=24):
+    """ GA4GH digest function """
+    digest = hashlib.sha512(seq.encode()).digest()
+    hex_digest = binascii.hexlify(digest[:offset])
+    return hex_digest.decode()
+```
 
-## Coordinate system as input
+### F6. Use cases for the `sorted_name_length_pairs` non-inherent attribute
 
-* **coordinate system -> seqcol digest**: If you have only a coordinate system (lengths with no sequences), you may still compute a seqcol digest using the seqcol algorithm, because the sequences themselves are optional. Coordinate systems can therefore be uniquely encoded, retrieved, and compared using the seqcol protocol.
+To illustrate the value of this attribute, we provide a short example. Consider a genome browser which allows users to display BED files identifying genomic loci of interest. The genome browser should only show BED files if they annotate the same reference genome coordinate system. This is looser than strict identity, since we don't really care what the underlying sequence characters are, as long as the positions are comparable. We also don't care about the order of the sequences. What we need can be fully defined logically by the level 1 digest of the `sorted_name_length_pairs` attribute.
 
+To check whether a BED file can be viewed for a particular genome, we simply compare the `sorted_name_length_pairs` digest of our reference genome with the sequence collection used to generate the file. There are only two possibilities for compatibility:
 
-## Related operations: What seqcol does *not* do
+2A) If the digests are equal, then the data file is directly compatible.
```

Comment:
Not sure why we define this as 2A and 2B, when they're the only ones available. Suggest 1/2 or A/B.